### PR TITLE
Enable bench unittests for CI performance testing

### LINKED ISSUE #161: Benchmark testing
The main idea is to prerelease benchmark tests to identify the regression of speed.
Bitcoin has a benchmark logic, we could adapt it to our needs, if necessary.
First we need to try to run the current implementation and look at the results to decide on additional improvements. 
I suggest placing the results in this issue

### PR DESCRIPTION
**Pull request by tawmaz**
- Refactor of SQLite initialization to make integration with unittests easier

- Add bench step to GitHub "makecheck" workflow

- Disable prevector bench test

- Port nano test bench framework from BitcoinCore

### CODE REVIEW

**Comment by andyoknen** [comment]
@tawmaz Are conflict fixes needed? Or is the work still underway?

**Comment by tawmaz** [comment]
Looks like the copyright string is the only conflict.  I will fix this as soon as we get 0.20.18 out so as to not create any branch problems for the release.

**Review by andyoknen — APPROVED** [review]

File: src/bench/bench_pocketcoin.cpp [hunk 1]
```diff
@@ -11,91 +11,144 @@
...
-        printer.reset(new benchmark::PlotlyPrinter(
-            gArgs.GetArg("-plot-plotlyurl", DEFAULT_PLOT_PLOTLYURL),
-            gArgs.GetArg("-plot-width", DEFAULT_PLOT_WIDTH),
-            gArgs.GetArg("-plot-height", DEFAULT_PLOT_HEIGHT)));
-    }
+    PocketDb::InitSQLite(GetDataDir() / "pocketdb");
 
-    benchmark::BenchRunner::RunAll(*printer, evaluations, scaling_factor, regex_filter, is_list_only);
+    // Go up two directories to access the checkpoints folder, assume we are running in /src/test
+    fs::path checkpointsPath = fs::system_complete("../..");
```
Comment (lostystyg) [inline src/bench/bench_pocketcoin.cpp:140]: Why not use same logic as for other nets? I mean searching for checkpoints in datadir. This assuming prevents to run bench tests if build tree differ from source that is a common situation.

**Review by lostystyg — APPROVED** [review]

File: src/bench/bench_pocketcoin.cpp (hunk 1 above)
Comment (andyoknen) [reply src/bench/bench_pocketcoin.cpp:140]: We will deal with this issue here https://github.com/pocketnetteam/pocketnet.core/issues/182

**Pull request merged**